### PR TITLE
Simplify models and implementation

### PR DESCRIPTION
**Pull request by danjoa**
Streamlines and minimises custom code in service impl, also by using generic validation through @mandatory and defaults

### CODE REVIEW

File: db/schema.cds [hunk 1]
```diff
@@ -12,43 +12,36 @@ namespace sap.fe.cap.travel;
...
   key TravelUUID : UUID;
   TravelID       : Integer @readonly default 0;
-  BeginDate      : Date;
-  EndDate        : Date;
-  BookingFee     : Decimal(16, 3);
-  TotalPrice     : Decimal(16, 3) @readonly;
-  CurrencyCode   : Currency;
+  BeginDate      : Date @mandatory;
+  EndDate        : Date @mandatory;
+  BookingFee     : Decimal(16,3) @mandatory default 11;
```
Comment (johannes-vogel) [inline db/schema.cds:17]: default and mandatory?

**Review by johannes-vogel — COMMENTED** [review]
Initially, we aligned this with ABAP implementation and they did the validation in custom code.
If we want to use the generic implementation, we should check Java failures.

File: db/schema.cds (hunk 1 above)
Comment (stewsk) [reply db/schema.cds:17]: @danjoa : what is the purpose of having both `@mandatory` and `default` for a Decimal value?

**Comment by stewsk** [comment]
@rjayasinghe : based on the new annotations, would you be able to also simplify the Java code like this was possible for Node?

File: db/schema.cds (hunk 1 above)
Comment (danjoa) [reply db/schema.cds:17]: True, makes no sense

Comment (danjoa) [inline db/schema.cds:17]:
```suggestion
  BookingFee     : Decimal(16,3) default 0;
```

**Comment by danjoa** [comment]
> @rjayasinghe : based on the new annotations, would you be able to also simplify the Java code like this was possible for Node?

Yes, we should do so

**Comment by rjayasinghe** [comment]
> > @rjayasinghe : based on the new annotations, would you be able to also simplify the Java code like this was possible for Node?
> 
> Yes, we should do so

We'll have a look. 👍🏽

**Comment by stewsk** [comment]
@rjayasinghe do you want to add your simplification changes (assuming there are any) to this PR or would you use a new one?

**Review by stewsk — APPROVED** [review]

**Pull request merged**